src/main.tsx

```diff
@@ -4,72 +4,98 @@
  */
 
 import { linkTo } from '@nextcloud/router'
-import { StrictMode, lazy } from 'react'
+import { StrictMode, lazy, Suspense } from 'react'
 import { createRoot } from 'react-dom'
+import { loadState } from '@nextcloud/initial-state'
 
 import './viewer.css'
 
 window.EXCALIDRAW_ASSET_PATH = linkTo('whiteboard', 'dist/')
 
-const Component = {
-	name: 'Whiteboard',
-	/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-	render(createElement: (arg0: string, arg1: { attrs: { id: string } }, arg2: string) => any) {
-		const App = lazy(() => import('./App'))
-		this.$emit('update:loaded', true)
-		const randomId = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(2, 10)
-		this.$nextTick(() => {
-			const rootElement = document.getElementById('whiteboard-' + randomId)
-			this.root = createRoot(rootElement)
+const App = lazy(() => import('./App'))
 
-			this.root.render(
-				<StrictMode>
-					<App fileId={this.fileid} isEmbedded={this.isEmbedded} fileName={this.basename} />
-				</StrictMode>,
-			)
+const generateRandomId = () => Math.random().toString(36).replace(/[^a-z]+/g, '').substr(2, 10)
+
+const renderApp = (rootElement, props) => {
+	const root = createRoot(rootElement)
+	root.render(
+		<StrictMode>
+			<Suspense fallback={<div>Loading...</div>}>
+				<App {...props} />
+			</Suspense>
+		</StrictMode>,
+	)
+	return root
+}
+
+const publicSharingToken = document.getElementById('sharingToken')?.value || null
+
+if (publicSharingToken) {
```

Comment:
We should also limit this to single file shares (see https://github.com/nextcloud/text/blob/main/src/public.js#L143-L151 how text checks for if a files table is rendered)